src/utils/utils.ts

```diff
@@ -26,20 +27,44 @@ function makeCanApi(options: CanCommunityOptions) {
  * @param community_account
  * @param code_id
  */
-async function findCode(code: EosName, community_account: EosName, code_id: CODE_IDS): Promise<any> {
-  const res = await app.rpc.get_table_rows({
-    code,
-    scope: community_account,
-    table: TableNameEnum.CODES,
-    lower_bound: code_id,
-    upper_bound: code_id,
-    index_position: 2,
-    key_type: 'i64',
-  });
+async function findCode(
+  code: EosName,
+  community_account: EosName,
+  code_id: CODE_IDS,
+  code_type: CodeTypeEnum,
+  referenceId?: number,
+): Promise<any> {
+  let res;
+  if (code_type === CodeTypeEnum.NORMAL || code_type === CodeTypeEnum.AMENDMENT) {
+    const codeTable = await app.rpc.get_table_rows({
+      code,
+      scope: community_account,
+      table: TableNameEnum.CODES,
+      lower_bound: code_id,
+      upper_bound: code_id,
+      index_position: 2,
+      key_type: 'i64',
+    });
+    res = codeTable?.rows[0];
+  } else if (code_type === CodeTypeEnum.POSITION) {
+    const codeTable = await app.rpc.get_table_rows({
+      code,
+      scope: community_account,
+      table: TableNameEnum.CODES,
+      lower_bound: referenceId,
+      upper_bound: referenceId,
+      index_position: 3,
+      key_type: 'i64',
+    });
+    const positionCodes = codeTable?.rows;
```

Comment:
why this function name is `findCode` but the param is specific to position-code only.

Reply:
It's not specific to position-code only, in the future it can be used to find badge-code.